src/test/compile-fail-fulldeps/proc-macro/span-api-tests.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// aux-build:span-api-tests.rs
+
+#![feature(proc_macro)]
+
+extern crate span_api_tests;
+
+use span_api_tests::{reemit, assert_no_source_file};
+
+reemit! {
+    assert_no_source_file! { "Hello, world!" }
```

Comment:
@jseyfried The `"Hello, world!"` token here gives its source file as `span-api-tests.rs` even though it was reparsed from a string by `reemit! { }`. Is this the correct behavior? If so, we can't rely on the source file being `"<anon>"` to determine if it was emitted from a macro or not. I think the output from external crates' macros is marked as `"<anon>"` though.